[Bug] tabletReport function of ReportHandler in fe throws NullPointerException

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.


### Version

master

### What's Wrong?

2022-03-14 20:42:21,668 ERROR (Thread-35|83) [Daemon.run():118] daemon thread got exception. name: Thread-35
java.lang.NullPointerException: null
        at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:603) ~[?:?]
        at java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:678) ~[?:?]
        at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:737) ~[?:?]
        at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159) ~[?:?]
        at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173) ~[?:?]
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233) ~[?:?]
        at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497) ~[?:?]
        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:661) ~[?:?]
        at org.apache.doris.catalog.TabletInvertedIndex.tabletReport(TabletInvertedIndex.java:134) ~[palo-fe.jar:3.4.0]
        at org.apache.doris.master.ReportHandler.tabletReport(ReportHandler.java:276) ~[palo-fe.jar:3.4.0]
        at org.apache.doris.master.ReportHandler.access$400(ReportHandler.java:95) ~[palo-fe.jar:3.4.0]
        at org.apache.doris.master.ReportHandler$ReportTask.exec(ReportHandler.java:240) ~[palo-fe.jar:3.4.0]
        at org.apache.doris.master.ReportHandler.runOneCycle(ReportHandler.java:1060) ~[palo-fe.jar:3.4.0]
        at org.apache.doris.common.util.Daemon.run(Daemon.java:116) [palo-fe.jar:3.4.0]
Caused by: java.lang.NullPointerException

### What You Expected?

not throw NullPointerException

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/fe/fe-core/src/main/java/org/apache/doris/catalog/TabletInvertedIndex.java
+++ b/fe/fe-core/src/main/java/org/apache/doris/catalog/TabletInvertedIndex.java
@@ -212,20 +212,8 @@ public class TabletInvertedIndex {
                                                     + "clear it from backend [{}]", transactionId, backendId);
                                         } else if (transactionState.getTransactionStatus() == TransactionStatus.VISIBLE) {
                                             TableCommitInfo tableCommitInfo = transactionState.getTableCommitInfo(tabletMeta.getTableId());
-                                            PartitionCommitInfo partitionCommitInfo = tableCommitInfo.getPartitionCommitInfo(partitionId);
-                                            if (partitionCommitInfo == null) {
-                                                /*
-                                                 * This may happen as follows:
-                                                 * 1. txn is committed on BE, and report commit info to FE
-                                                 * 2. FE received report and begin to assemble partitionCommitInfos.
-                                                 * 3. At the same time, some of partitions have been dropped, so partitionCommitInfos does not contain these partitions.
-                                                 * 4. So we will not able to get partitionCommitInfo here.
-                                                 *
-                                                 * Just print a log to observe
-                                                 */
-                                                LOG.info("failed to find partition commit info. table: {}, partition: {}, tablet: {}, txn id: {}",
-                                                        tabletMeta.getTableId(), partitionId, tabletId, transactionState.getTransactionId());
-                                            } else {
+                                            PartitionCommitInfo partitionCommitInfo = tableCommitInfo == null ? null : tableCommitInfo.getPartitionCommitInfo(partitionId);
+                                            if (partitionCommitInfo != null) {
                                                 TPartitionVersionInfo versionInfo = new TPartitionVersionInfo(tabletMeta.getPartitionId(),
                                                         partitionCommitInfo.getVersion(), 0);
                                                 synchronized (transactionsToPublish) {

